docs/standard/native-interop/abi-support.md

```diff
@@ -0,0 +1,82 @@
+
+The COM and `IUnknown` ABI was defined to align with the C language. It was specifically designed to support Object Oriented Programming, similar to C++, but to provide a stable ABI. .NET has a deep history with COM and `IUnknown` and as this ABI was originally designed to align with C, it is supported on all .NET platforms.
+
+In .NET 5+, low-level, cross-platform, `IUnknown` lifetime support is provided by <xref:System.Runtime.InteropServices.ComWrappers>. In .NET 8+, <xref:System.Runtime.InteropServices.Marshalling.GeneratedComInterfaceAttribute> and <xref:System.Runtime.InteropServices.Marshalling.GeneratedComClassAttribute> provide source generated C# projections. When targeting versions prior to .NET 5, the [built-in COM interop system](/dotnet/standard/native-interop/cominterop) must be used and is limited to the Windows platforms.
+
+The WinRT platform represents an evolution of the COM and `IUnknown` ABI. Support for this is provided by the [CsWinRT toolkit](/windows/apps/develop/platform/csharp-winrt/) and is built upon <xref:System.Runtime.InteropServices.ComWrappers>.
+
+Additional links:
+
+* [`ComWrappers` walkthrough](/dotnet/standard/native-interop/tutorial-comwrappers)
+* [COM source generator sample](/dotnet/standard/native-interop/comwrappers-source-generation)
+* [ClangSharp](https://github.com/dotnet/ClangSharp) binding generator
+
+## Java Virtual Machine (JVM) based languages
+
+The Java Virtual Machine (JVM) defines a foreign function interface (FFI) in C to interoperate with other platforms. Interoperability between .NET and Java can be achieved via this interface.
+
+Additional links:
+
+* [Java binding generator](https://github.com/dotnet/java-interop)
+
+## Swift
+
+The Swift programming environment has a well-defined stable ABI that is [supported in .NET](https://github.com/dotnet/designs/blob/main/proposed/swift-interop.md). In .NET 9+, specific APIs that support interop with Swift can be found under the <xref:System.Runtime.InteropServices.Swift> namespace.
+
+## Objective-C
+
+The Objective-C language follows the C language's ABI and is [supported in .NET](https://github.com/dotnet/designs/blob/main/accepted/2021/objectivec-interop.md). In .NET 8+, specific APIs that support interop with Objective-C can be found under the <xref:System.Runtime.InteropServices.ObjectiveC> namespace.
+
+Additional links:
+
+* [Objective-Sharpie](/previous-versions/xamarin/cross-platform/macios/binding/)
+* [Objective-C binding walkthrough](/dotnet/maui/migration/ios-binding-projects)
+
+## Python
+
+The reference implementation of the Python run-time, [CPython](https://github.com/python/cpython), defines a foreign function interface (FFI) in C to interoperate with other platforms. Interoperability between .NET and Python can be achieved via this interface.
+
+Additional links:
+
+* [Providing a C API for an Extension Module](https://docs.python.org/3/extending/extending.html#providing-a-c-api-for-an-extension-module)
```

Comment:
Mention https://github.com/pythonnet/pythonnet and https://github.com/tonybaloney/csnakes?

Reply:
I was going to avoid the csnakes because it is so new and still very much in flux so I'm inclined to wait a bit. I will mention pythonnet.